roles/iam-user/tasks/create_groups.yml

```diff
@@ -1,15 +1,29 @@
 ---
 
-- name: Create developer IAM group
+- name: Create IAM groups
   iam:
     state: present
     iam_type: group
-    name: "{{ iam_user_developer_groupname }}"
+    name: "{{ item.groups }}"
+  with_items: "{{ iam_user_users | default([]) }}"
```

Comment:
The name key here doesn't take a list. I will preprocess and set a var so it doesn't loop unnecessarily.